ddtrace/tracer/textmap.go

```diff
@@ -391,9 +393,10 @@ func (p *propagator) extractTextMap(reader TextMapReader) (ddtrace.SpanContext,
 		return nil, err
 	}
 	if ctx.trace != nil {
-		ctx.traceID128 = ctx.trace.propagatingTags[keyTraceID128]
+		// TODO: this always assumed it was valid so I copied that logic here, maybe we shouldn't
```

Comment:
So previously we just took in the string and propagated it, blindly ignoring if it was valid hex (It seems, I might be missing something though). So I carried forward that functionality here to try to keep this a "true refactor", but now that we're actually parsing that incoming upper bits if they're invalid we won't be able to pass them along (e.g. if someone sends a _dd.p.tid of `HELLOFRIENDS` previously I believe we would propagate that along, but now since it's invalid hex it won't actually be stored in the traceID and I don't believe will be propagated to further services)

Reply:
Gotcha. Might be worth a message on the internal channel just to clarify if we need to do this.